src/Dapr.Client/DaprClientGrpc.cs

```diff
@@ -1373,6 +1375,537 @@ private Autogenerated.StateItem ToAutogeneratedStateItem(StateTransactionRequest
+
+            //Send the plaintext bytes in 4 kb blocks in subsequent messages
+            const int blockSize = 4 * 1024; // 4 kb
+            await using (var bufferedStream = new BufferedStream(stream, blockSize))
+            {
+                var buffer = new byte[blockSize];
+                int bytesRead;
+                ulong sequenceNumber = 0;
+
+                while ((bytesRead = await bufferedStream.ReadAsync(buffer, 0, blockSize, cancellationToken)) != 0)
+                {
+                    await duplexStream.RequestStream.WriteAsync(new Autogenerated.EncryptRequest
+                    {
+                        Payload = new Autogenerated.StreamPayload
+                        {
+                            Data = ByteString.CopyFrom(buffer, 0, bytesRead), Seq = sequenceNumber
+                        }
+                    }, cancellationToken);
+
+                    //Increment the sequence number
+                    sequenceNumber++;
+                }
+            }
+            
+            //Send the completion message
+            await duplexStream.RequestStream.CompleteAsync();
+
+            //Set up how the response is handled since we know it's of finite length
+            var returnedBytes = new List<byte>();
+            await foreach (var encryptResponse in duplexStream.ResponseStream.ReadAllAsync(cancellationToken).ConfigureAwait(false))
+            {
+                returnedBytes.AddRange(encryptResponse.Payload.Data.ToByteArray());
+            }
+
+            return returnedBytes.ToArray();
+        }
+
+        /// <inheritdoc />
+        [Obsolete("The API is currently not stable as it is in the Alpha stage. This attribute will be removed once it is stable.")]
+        public override Task<byte[]> EncryptAsync(
+            string vaultResourceName, Stream plainTextStream, KeyWrapAlgorithm algorithm,
+            string keyName, DataEncryptionCipher dataEncryptionCipher = DataEncryptionCipher.AesGcm,
+            CancellationToken cancellationToken = default) => EncryptStreamAsync(vaultResourceName, plainTextStream,
+            algorithm, keyName, dataEncryptionCipher, string.Empty, true, cancellationToken);
+
+        /// <inheritdoc />
+        [Obsolete("The API is currently not stable as it is in the Alpha stage. This attribute will be removed once it is stable.")]
+        public override Task<byte[]> EncryptAsync(
+            string vaultResourceName, Stream plainTextStream, KeyWrapAlgorithm algorithm,
+            string keyName, string decryptionKeyName,
+            DataEncryptionCipher dataEncryptionCipher = DataEncryptionCipher.AesGcm,
+            CancellationToken cancellationToken = default) => EncryptStreamAsync(vaultResourceName, plainTextStream,
+            algorithm, keyName, dataEncryptionCipher,
+            decryptionKeyName, false, cancellationToken);
+
+        private async Task<byte[]> EncryptByteArrayAsync(string vaultResourceName, byte[] plainTextBytes,
```

Comment:
Could we not just wrap `plainTextBytes` in a `MemoryStream` and then use a single stream-based encrypt implementation?

Reply:
Why, yes.. Yes we could. Added in the latest commit.